### PR TITLE
Move extratest function into main_common

### PR DESCRIPTION
**Pull request by asmorodskyi**
Remove code duplication and trying to minimize quantity of tests
 which running only in one side(opensuse/osd).
Related ticket #poo16020

### CODE REVIEW

**Comment by asmorodskyi** [comment]
Prove runs 
http://kimball.arch.suse.de/tests/287
http://kimball.arch.suse.de/tests/288
http://kimball.arch.suse.de/tests/289
http://kimball.arch.suse.de/tests/290

**Comment by okurz** [comment]
Please reference full URLs or markdown syntax for ticket references in PR messages and/or commit messages, e.g. either
https://progress.opensuse.org/issues/16020 or [poo#16020](https://progress.opensuse.org/issues/16020)

**Comment by asmorodskyi** [comment]
@okurz , @mnowaksuse , @kravciak

**Comment by okurz** [comment]
Unfortunately you lost now again recent changes introduced by @drpaneas. I know it's hard. Please try one more time and we will try to not merge any extratest related changes before this is done -> @Soulofdestiny

File: lib/main_common.pm [hunk 1]
```diff
@@ -406,4 +407,97 @@ sub maybe_load_kernel_tests {
...
+        if (!get_var("OFW") && !is_jeos) {
+            loadtest "console/aplay";
+        }
+        if (get_var("FILESYSTEM", "btrfs") eq "btrfs") {
+            loadtest "console/btrfs_autocompletion";
+            if (get_var("NUMDISKS", 0) > 1) {
+                loadtest "console/btrfs_qgroups";
+                loadtest "console/btrfs_send_receive";
+                if (check_var('DISTRI', 'sle')) {
+                    loadtest 'console/snapper_cleanup';
```
Comment (mnowaksuse) [inline lib/main_common.pm]: It used to be: `if (sle_version_at_least('12-SP2')) {`. The test would probably fail on pre-12-SP2?

**Comment by asmorodskyi** [comment]
I merge @drpaneas changes to my PR . @drpaneas  you can double check , also I refactor them on the fly. If you not agree or it's not clear for you fell free to ping me

File: lib/main_common.pm (hunk 1 above)
Comment (asmorodskyi) [inline lib/main_common.pm]: You right , fixed

**Comment by drpaneas** [comment]
LGTM

File: lib/main_common.pm [hunk 2]
```diff
@@ -406,4 +407,116 @@ sub maybe_load_kernel_tests {
...
+sub load_extra_tests() {
+    # Put tests that filled the conditions below
+    # 1) you don't want to run in stagings below here
+    # 2) the application is not rely on desktop environment
+    # 3) running based on preinstalled image
+    return unless get_var('EXTRATEST');
+    # pre-conditions for extra tests ie. the tests are running based on preinstalled image
+    return if get_var("INSTALLONLY") || get_var("DUALBOOT") || get_var("RESCUECD");
+
+    if (run_aggregated_maintenance_tests()) {
```
Comment (okurz) [inline lib/main_common.pm]: good idea

**Pull request merged**